lib/realtime/vehicle.ex

```diff
@@ -199,31 +204,87 @@ defmodule Realtime.Vehicle do
     end
   end
 
-  @spec scheduled_timepoint_status([StopTime.t()], Util.Time.timestamp()) ::
-          timepoint_status() | nil
-  def scheduled_timepoint_status([], _now) do
+  @doc """
+  If a block isn't scheduled to have started yet:
+    the start of the first trip
+  If a block is scheduled to have finished:
+    the end of the last trip
+  If now is in the middle of a layover:
+    the end of the previous trip
+  If now is in the middle of a trip:
+    the next timpeoint in that trip
+  """
+  @spec scheduled_location(Block.t(), Util.Time.timestamp()) :: scheduled_location() | nil
+  def scheduled_location(nil, _now) do
     nil
   end
 
-  def scheduled_timepoint_status(stop_times_on_trip, now) do
-    trip_start = List.first(stop_times_on_trip).time
-    now_time_of_day = Util.Time.nearest_time_of_day_for_timestamp(now, trip_start)
-    timepoints = Enum.filter(stop_times_on_trip, &is_a_timepoint?/1)
+  def scheduled_location([], _now) do
+    nil
+  end
 
-    case Realtime.Helpers.find_and_previous(timepoints, fn timepoint ->
-           timepoint.time >= now_time_of_day
-         end) do
-      nil ->
-        nil
+  def scheduled_location(block, now) do
+    block_start = List.first(List.first(block).stop_times).time
+    block_end = List.last(List.last(block).stop_times).time
+    now_time_of_day = Util.Time.nearest_time_of_day_for_timestamp(now, block_start)
+
+    trip =
```

Comment:
Could this be broken out into a private function? Same with `timepoint_status` assignment below. This function is pretty long, so I think that any way of condensing what is happening might be helpful.